sdk/opendp/smartnoise/sql/private_rewriter.py

```diff
@@ -182,19 +191,10 @@ def query(self, query):
     def exact_aggregates(self, query):
         child_scope = Scope()
 
-        if self.options.row_privacy:
-            keycount_expr = AggFunction(FuncName("COUNT"), None, AllColumns())
-        else:
-            key_col = self.key_col(query)
-            keycount_expr = AggFunction(FuncName("COUNT"), Token("DISTINCT"), Column(key_col))
-            child_scope.push_name(keycount_expr.expression)
-
         for ne in query.select.namedExpressions:
             child_scope.push_name(ne.expression)
 
-        keycount = NamedExpression(Identifier("keycount"), keycount_expr)
-
-        select = [keycount] + [ne for ne in query.select.namedExpressions]
```

Comment:
Prepending `[keycount]` meant that the keycount was not registered in the child scope of the outer call (`query()`), which prevented the rewriter from reusing keycount expressions.